cscs-checks/mch/alltoallv.py

```diff
@@ -0,0 +1,93 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.parameterized_test([''], ['--nocomm'], ['--nocomp'])
+class Alltoallv(rfm.RegressionTest):
+    def __init__(self, exec_parameter):
+        super().__init__()
+        self.valid_systems = ['daint:gpu', 'dom:gpu', 'kesch:cn']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        if self.current_system.name in ['daint', 'dom']:
+            self.modules = ['craype-accel-nvidia60']
+            self._pgi_flags = ['-acc', '-ta=tesla:cc60', '-Mnorpath']
+        elif self.current_system.name in ['kesch']:
+            self.modules = ['craype-accel-nvidia35']
+            self._pgi_flags = ['-O2', '-ta=tesla,cc35,cuda8.0']
+
+        self.num_tasks = 144
+        self.num_gpus_per_node = 16
+        self.num_tasks_per_node = 16
+        self.num_tasks_per_socket = 8
+        self.executable = ('build/src/comm_overlap_benchmark '
+                           '%s' % exec_parameter)
+        self.sourcesdir = 'https://github.com/cosunae/comm_overlap_bench'
+        # Checkout to the branch alltoallv
+        self.prebuild_cmd = ['git checkout alltoallv']
+        self.build_system = 'CMake'
+        self.build_system.builddir = 'build'
+        self.build_system.config_opts = ['-DMPI_VENDOR=mvapich2',
+                                         '-DCUDA_COMPUTE_CAPABILITY="sm_37"',
+                                         '-DCMAKE_BUILD_TYPE=Release',
+                                         '-DENABLE_MPI_TIMER=ON']
+        self.build_system.max_concurrency = 1
+        self.sanity_patterns = sn.assert_found(r'ELAPSED TIME:', self.stdout)
+        self.perf_patterns = {
+            'elapsed_time':
+            sn.extractsingle(r'ELAPSED TIME:\s+(?P<elapsed_time>\S+)',
+                             self.stdout, 'elapsed_time', float, 1)
+        }
+
+        if not exec_parameter:
+            ref = 5.53777
+        elif exec_parameter == '--nocomm':
+            ref = 5.7878
+        elif exec_parameter == '--nocomp':
+            ref = 5.62155
+
+        self.reference = {
+            'kesch:cn': {
+                'elapsed_time': (ref, None, 0.15)
+            },
```

Comment:
We need a reference for Daint/Dom, otherwise we simply state that this test is not supported for these systems and go away with that.